executor/habitat.go

```diff
@@ -1,40 +1,98 @@
 package executor
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/screwdriver-cd/sd-cmd/config"
+	"github.com/screwdriver-cd/sd-cmd/screwdriver/store"
 	"github.com/screwdriver-cd/sd-cmd/util"
 )
 
-var habPath = "/hab/bin/hab"
+const habPath = "/hab/bin/hab"
 
 // Habitat is the Habitat Executor struct
 type Habitat struct {
-	Args []string
-	Spec *util.Habitat
+	Args  []string
+	Spec  *util.CommandSpec
+	Store store.Store
 }
 
 // NewHabitat returns the Habitat struct
 func NewHabitat(spec *util.CommandSpec, args []string) (habitat *Habitat, err error) {
+	storeapi := store.New(config.SDStoreURL, spec, config.SDToken)
+
 	habitat = &Habitat{
-		Args: args,
-		Spec: &util.Habitat{
-			Mode:    spec.Habitat.Mode,
-			Package: spec.Habitat.Package,
-			Command: spec.Habitat.Command,
-		},
+		Args:  args,
+		Spec:  spec,
+		Store: storeapi,
 	}
 	return habitat, nil
 }
 
+func (h *Habitat) getPkgDirPath() string {
+	return filepath.Join(config.BaseCommandPath, h.Spec.Namespace, h.Spec.Name, h.Spec.Version)
+}
+
+func (h *Habitat) getPkgFilePath() string {
+	fileName := filepath.Base(h.Spec.Habitat.File)
+	return filepath.Join(h.getPkgDirPath(), fileName)
+}
+
+func (h *Habitat) isDownloaded() bool {
+	fInfo, err := os.Stat(h.getPkgFilePath())
+	if err != nil {
```

Comment:
There are many kind of error are there and some error is not error which you want. 
(e.g. EEXIST, ENOTDIR)
https://github.com/golang/go/blob/master/src/syscall/tables_nacljs.go#L99

Also if you see the method `isExist`, there is possibility that error is `exist` error.
https://github.com/golang/go/blob/master/src/os/error_plan9.go#L8

So I think it might be better to use os.IsNotExist().

Reply:
Is it better to use isExist() and check err nil
```
if err != nil && !isExist(err) {
   return false
}
```

Reply:
I understand what you mean. I will fix like that, thank you!